apps/web/src/components/activity/ExecutionDetailsStepContent.tsx

```diff
@@ -52,19 +53,21 @@ export const ExecutionDetailsStepContent = ({ step }) => {
               <ExecutionDetail
                 key={`execution-detail-${executionDetail.id}`}
                 executionDetail={executionDetail}
-                showTriggerSnippet={showTriggerSnippet}
-                onShowExecutionDetail={onShowExecutionDetail}
+                showTriggerSnippet={detailId !== executionDetail.id}
+                onShowExecutionDetail={onShowExecutionDetail(executionDetail.id, executionDetail.raw)}
```

Comment:
My React might be a bit rusty but isn't this going to get executed? Shouldn't we pass a function declaration?
```suggestion
                onShowExecutionDetail={() => onShowExecutionDetail(executionDetail.id, executionDetail.raw)}
```

Reply:
We do check how onShowExecutionDetail is implemented 😄